controller/RestController.php

```diff
@@ -683,18 +683,30 @@ public function label($request)
             return $this->returnError(400, "Bad Request", "uri parameter missing");
         }
 
-        $results = $request->getVocab()->getConceptLabel($request->getUri(), $request->getLang());
-        if ($results === null) {
+        $labelResults = $request->getVocab()->getAllConceptLabels($request->getUri(), $request->getLang());
+        if ($labelResults === null) {
             return $this->returnError('404', 'Not Found', "Could not find concept <{$request->getUri()}>");
         }
 
         $ret = array_merge_recursive($this->context, array(
-            '@context' => array('prefLabel' => 'skos:prefLabel', '@language' => $request->getLang()),
+            '@context' => array('prefLabel' => 'skos:prefLabel', 'altLabel' => 'skos:altLabel', 'hiddenLabel' => 'skos:hiddenLabel', '@language' => $request->getLang()),
             'uri' => $request->getUri())
         );
 
-        if (isset($results[$request->getLang()])) {
-            $ret['prefLabel'] = $results[$request->getLang()]->getValue();
+        if (!empty($labelResults['prefLabel'])) {
+            $ret['prefLabel'] = $labelResults['prefLabel'][0]->getValue();
```

Comment:
I think the `->getValue()` call is in the wrong layer here. It should be already inside GenericSparql, so that there is no need to pass EasyRdf Literal objects from the model all the way to the controller.

Reply:
Also, if there are no `getValue()` calls here, then there should also be no need for the `if (!empty(...))` checks and copying of arrays. `$labelResults` can just be returned directly I think

Reply:
Good point. I changed it to pass only stings for the controller. I suppose the empty checks are sort of unnecessary - my design intent was to only give out data that is there:

```
"uri":"http://www.yso.fi/onto/yso/p1191",
"prefLabel":"aallot",
"hiddenLabel":["aalto"]
```

instead of:
```
"uri":"http://www.yso.fi/onto/yso/p1191",
"prefLabel":"aallot",
"altLabel":[] (or "altLabel": null)
"hiddenLabel":["aalto"]
```

But I suppose is is sort of explicit in the latter example that the method tried to find alt labels, but didn't find them. I'll make the changes.

Reply:
I understand the intent. But this is already handled by the `foreach` loop. If there are no hiddenLabels in the result set, there will be no `hiddenLabel` key in the array since it will never get added. No need for an explicit `if (!count(...))` check.

Reply:
True. I realised the simplest thing to do was reassigning the value of prefLabel key, and otherwise just merging the $labelResult into the $ret.